### ISSUE TITLE
Broken pipe error (as reported in logs) when trying to generate image styles in a Drupal 8 project

### ISSUE THREAD

**Issue by cscomfort** [drud/ddev#2928]
https://gist.github.com/cscomfort/c9d077eba473a28056134dd32498bf26

**Describe the bug**
I'm not sure if this a Drupal 8 (v8.9.12) issue or DDEV, but when flushing thumbnails in my project `drush if --all` and then visiting a page that should regenerate said thumbnails, the process fails. The associated image links fail as 404s (their paths are rooted at `/var/www/html/web/sites/default/files/styles` as seen in the container). Watching the logs, I see the following when visiting the pages that should regenerate the thumbnails (seen here is a 'test' image style):

`2021/04/05 18:40:54 [info] 1670#1670: *823 writev() failed (32: Broken pipe) while sending to client, client: 172.18.0.6, server: _, request: "GET /sites/default/files/styles/test/public/images/2021-01/woman-scientist-with-fruit-in-lab.jpg HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm.sock:", host: "local.utni.org", referrer: "http://local.utni.org/admin/content/media"`

Note that I don't encounter any issue retrieving images on the public side of the site.

**To Reproduce**
I deleted (`ddev delete`) and then started the project anew, to no avail.

**Expected behavior**
Images styles should be generated.

**Version and configuration information (please complete the following information):**

- Host computer OS and Version: macOS Big Sur
- Docker Desktop version: 3.10
- ddev version information: v1.16.7
- config.yaml contents for the misbehaving project:
```yaml
name: utni-v3
type: drupal8
docroot: web
php_version: "7.3"
webserver_type: nginx-fpm
router_http_port: "80"
router_https_port: "443"
xdebug_enabled: false
additional_hostnames: []
additional_fqdns: [
    'local.utni.org'
]
provider: default
use_dns_when_possible: true
```
I disabled the one custom yaml config which simply provided a 'local' environment variable. The issue continues to persist.

Thanks in advance!

**Comment by rfay** [comment]
It's questionable using `additional_fqdns` that can be masked by real hostnames, and not recommended. Better `additional_hostnames: [utnilocal]`, which would give you utnilocal.ddev.site. But that's not going to be your problem.

You casually mention using drush. Were you using it inside the web container? That's where you want to use it. `ddev ssh` and use `drush if --all` or use `ddev drush if --all`.   

Please try your recreation scenario with a trivial site, like a plain vanilla umami install, per https://ddev.readthedocs.io/en/latest/users/cli-usage/#drupal-8-quickstart or however you want to do it. See if you can get it to misbehave on a site like that. I think it won't misbehave. Then try to look at what extensions or configuration your "real" site may have going.

**Comment by rfay** [comment]
A race issue that affects multiple versions of Drupal was reported in [drupal slack](https://drupal.slack.com/archives/C5TQRQZRR/p1617730183250300), https://www.drupal.org/project/drupal/issues/3036494

Might that be related to your problem?

**Comment by cscomfort** [comment]
Resolved: one of my config settings in <code>settings.local.php</code> was the culprit:

```php
$config['image.settings']['suppress_itok_output'] = TRUE;
```

I had forgotten that I had enabled this option. It was functioning as desired on the frontend, but since it had been quite some time since I uploaded an image in the admin I did not notice the issue image style issue. It seems that -- for my environment at least -- it's also necessary to enable the following:

```php
$config['image.settings']['allow_insecure_derivatives'] = TRUE;
```

Thanks for taking the time to look into it.
